tests/apichecks/crds_test.go

```diff
@@ -242,3 +242,33 @@ func visitProps(props *apiextensions.JSONSchemaProps, fieldPath string, callback
 		klog.Fatalf("unhandled props.Type %q in %+v", props.Type, props)
 	}
 }
+
+func TestCRDCamelCase(t *testing.T) {
+	crds, err := crdloader.LoadAllCRDs()
+	if err != nil {
+		t.Fatalf("error loading crds: %v", err)
+	}
+	var errs []string
+	for _, crd := range crds {
+		for _, version := range crd.Spec.Versions {
+			visitCRDVersion(version, func(field *CRDField) {
+				fieldPath := field.FieldPath
+				first := func() int32 {
+					tokens := strings.Split(fieldPath, ".")
+					// Only check the last token to avoid duplication.
```

Comment:
Not sure duplication is the end of the world here, but works for me!

Reply:
oh, that will end up as giving us something below if  the pascal case happens in the second substring place. 
```
 field ".spec.PrivatePoolV1Config" should use camel case
 field ".spec.PrivatePoolV1Config.networkConfig" should use camel case
 field ".spec.PrivatePoolV1Config.networkConfig.peeredNetworkIPRange" should use camel case
```

Reply:
What you have done is better :-)